generic/tests/lvm.py

```diff
@@ -44,55 +55,57 @@ def run(test, params, env):
-                      logging.info)
-        session.cmd("vgcreate %s %s" % (vg_name, disks))
-
-        error.context("activating volume group %s" % vg_name)
-        session.cmd("vgchange -ay %s" % vg_name)
-
-        error.context("creating logical volume on volume group %s" % vg_name,
-                      logging.info)
-        session.cmd("lvcreate -L2000 -n %s %s" % (lv_name, vg_name))
-
-        error.context(
-            "creating ext3 filesystem on logical volume %s" % lv_name)
-        session.cmd("yes | mkfs.ext3 %s" % lv_path, timeout=int(timeout))
-
-        mount_lv(lv_path, session)
-
-        umount_lv(lv_path, session)
-
-        error.context("checking ext3 filesystem made on logical volume %s" %
-                      lv_name, logging.info)
-        session.cmd("fsck %s" % lv_path, timeout=int(timeout))
-
-        if clean == "no":
+        if sub_type == "lvm_create":
+            disk_list = []
+            for disk in params.objects("images")[-2:]:
+                d_id = params["blk_extra_params_%s" % disk].split("=")[1]
+                d_path = utils_misc.get_linux_drive_path(session, d_id)
+                if not d_path:
+                    raise error.TestError("Failed to get '%s' drive path"
+                                          % d_id)
+                disk_list.append(d_path)
+            disks = " ".join(disk_list)
+            error.context("adding physical volumes %s" % disks, logging.info)
+            session.cmd("pvcreate %s" % disks)
+            error.context("creating a volume group out of %s" % disks,
+                          logging.info)
+            session.cmd("vgcreate %s %s" % (vg_name, disks))
+            error.context("activating volume group %s" % vg_name,
+                          logging.info)
+            session.cmd("vgchange -ay %s" % vg_name)
+            error.context("creating logical volume on volume group %s"
+                          % vg_name, logging.info)
+            session.cmd("lvcreate -L2000 -n %s %s" % (lv_name, vg_name))
+            error.context("creating ext3 filesystem on logical volume %s"
+                          % lv_name, logging.info)
+            session.cmd("yes | mkfs.ext3 %s" % lv_path, timeout=int(timeout))
             mount_lv(lv_path, session)
-
+            umount_lv(lv_path, session)
+            error.context("checking ext3 filesystem made on logical volume %s"
+                          % lv_name, logging.info)
+            session.cmd("fsck %s" % lv_path, timeout=int(timeout))
+            if clean == "no":
+                mount_lv(lv_path, session)
+        elif sub_type == "fillup_disk" or sub_type == "ioquit":
+            if not check_mount_lv(check_mount, session):
+                mount_lv(lv_path, session)
+            utils_test.run_virt_sub_test(test, params, env, sub_type)
+        elif sub_type == "lvm_clean":
+            pass
+        else:
+            raise error.TestError("Failed to get sub_type")
```

Comment:
Module `autotest.client.shared.error` should not be referenced any more, please use `test.error() | test.fail()` instead of raising the error here, and use `virttest.error_context.context()` instead of `error.context()`.